src/azul/collections.py

```diff
@@ -124,3 +127,37 @@ def compose_keys(f, g):
 
     """
     return lambda v: f(g(v))
+
+
+def none_safe_itemgetter(*items):
+    """
+    Like `itemgetter` except that the returned callable returns `None`
+    (or a tuple of `None`) if it's passed None.
+
+    >>> f = none_safe_itemgetter('foo', 'bar')
+    >>> f({'foo': 1, 'bar': 2})
+    (1, 2)
+
+    >>> f(None)
+    (None, None)
+
+    >>> none_safe_itemgetter('foo')({'foo':123})
+    123
+
+    >>> none_safe_itemgetter('foo')(None) is None
+    True
+
+    """
+
+    if len(items) > 1:
+        n = (None,) * len(items)
+    elif len(items) == 1:
+        n = None
+    else:
+        assert False
```

Comment:
Should behave identically to invoking `itemgetter()` without arguments. Also need doctest for that.